mantidimaging/core/operations/median_filter/median_filter.py

```diff
@@ -133,36 +145,9 @@ def modes():
     return ['reflect', 'constant', 'nearest', 'mirror', 'wrap']
 
 
-def _median_filter(data: np.ndarray, size: int, mode: str):
-    # Replaces NaNs with negative infinity before median filter
-    # so they do not effect neighbouring pixels
+def _median_filter(data: np.ndarray, size: int, mode: str) -> np.ndarray:
     nans = np.isnan(data)
     data = np.where(nans, -np.inf, data)
     data = scipy_ndimage.median_filter(data, size=size, mode=mode)
-    # Put the original NaNs back
     data = np.where(nans, np.nan, data)
     return data
```

Comment:
It would be good to keep the comments here as they explain why it does odd things with the nans.